components/organisms/o-personal-details.vue

```diff
@@ -0,0 +1,258 @@
+        :label="$t('First name *')"
+        :valid="!$v.personalDetails.firstName.$error"
+        :error-message="
+          !$v.personalDetails.firstName.required
+            ? $t('Field is required')
+            : !$v.personalDetails.firstName.minLength
+              ? $t('Name must have at least 2 letters.')
+              : ''
+        "
+        @blur="$v.personalDetails.firstName.$touch()"
+      />
+      <SfInput
+        v-model.trim="personalDetails.lastName"
+        class="form__element form__element--half form__element--half-even"
+        name="last-name"
+        :label="$t('Last name *')"
+        :valid="!$v.personalDetails.lastName.$error"
+        :error-message="
+          !$v.personalDetails.lastName.required ? $t('Field is required') : ''
+        "
+        @blur="$v.personalDetails.lastName.$touch()"
+      />
+      <SfInput
+        v-model.trim="personalDetails.emailAddress"
+        class="form__element"
+        name="email-address"
+        :label="$t('Email address *')"
+        :valid="!$v.personalDetails.emailAddress.$error"
+        :error-message="
+          !$v.personalDetails.emailAddress.required
+            ? $t('Field is required')
+            : !$v.personalDetails.emailAddress.email
+              ? $t('Please provide valid e-mail address.')
+              : ''
+        "
+        @blur="$v.personalDetails.emailAddress.$touch()"
+      />
+      <div class="form__element form__group">
+        <SfCheckbox
+          v-if="!currentUser"
+          v-model="createAccount"
+          class="form__element--half form__checkbox"
+          name="createAccount"
+        >
+          <template #label>
+            <span class="sf-checkbox__label">
+              {{ $t("I want to create an account") }}
+            </span>
+            <AIconAccountBenefits class="ml20" />
+          </template>
+        </SfCheckbox>
+      </div>
+      <template v-if="createAccount && !currentUser">
+        <SfInput
+          v-model="password"
+          type="password"
+          class="form__element"
+          name="password"
+          :label="$t('Password *')"
+          :valid="!$v.password.$error"
+          :error-message="
+            !$v.password.required
+              ? $t('Field is required')
+              : !$v.password.minLength
+                ? $t('Password must have at least 8 letters.')
+                : !$v.password.complex
+                  ? $t(
+                    'Password must contain at least 3 different character classes: lower case, upper case, digits, special characters.'
+                  )
+                  : ''
+          "
+          @blur="$v.password.$touch()"
+        />
+        <SfInput
+          v-model="rPassword"
+          type="password"
+          class="form__element"
+          name="password-confirm"
+          :label="$t('Repeat password *')"
+          :valid="!$v.rPassword.$error"
+          :error-message="
+            !$v.rPassword.required
```

Comment:
We don't need to check second condition and this can be typed simpler: `$v.fieldName.requred ? $t('Field is required') : $t('Other message')`

Reply:
Done.